tests/eosio.distribute_tests.cpp

```diff
@@ -0,0 +1,158 @@
+    public:
+        abi_serializer abi_dist_ser;
+
+        eosio_distribute_tester() : eosio_system_tester() {
+            produce_block();
+            // set eosio.dist code
+            set_code( dist_acct, contracts::distribute_wasm());
+            set_abi( dist_acct, contracts::distribute_abi().data() );
+
+            
+            const auto& accnt = control->db().get<account_object,by_name>( dist_acct );
+            abi_def abi;
+            BOOST_REQUIRE_EQUAL(abi_serializer::to_abi(accnt.abi, abi), true);
+            abi_dist_ser.set_abi(abi, abi_serializer::create_yield_function(abi_serializer_max_time));
+        }
+
+        action_result push_action(name account, name act, name authorizer, const variant_object& data) {
+            try {
+                base_tester::push_action(account, act, authorizer, data, 1);
+            } catch (const fc::exception& ex) {
+                edump((ex.to_detail_string()));
+                return error(ex.top_message());
+            }
+            return success();
+        }
+
+        action_result setdistrib(name authorizer, const std::vector<distribute_account>& accounts) {
+            return push_action(dist_acct, N(setdistrib), authorizer, mvo()("accounts", accounts));
+        }
+
+        action_result claimdistrib(name acct) {
+            return push_action(dist_acct, N(claimdistrib), acct, mvo()("claimer", acct));
+        }
+
+        fc::variant get_state(){
+            vector<char> data = get_row_by_account(dist_acct, dist_acct, N(state), N(state));
+            return data.empty() ? fc::variant() : abi_dist_ser.binary_to_variant("distribute_state", data, abi_serializer::create_yield_function(abi_serializer_max_time));
+        }
+
+        fc::variant get_claimer(const name& acct){
+            vector<char> data = get_row_by_account(dist_acct, dist_acct, N(claimers), acct);
+            return data.empty() ? fc::variant() : abi_dist_ser.binary_to_variant("distribute_claimer", data, abi_serializer::create_yield_function(abi_serializer_max_time));
+        }
+
+}; // eosio_distribute_tester
+
+BOOST_AUTO_TEST_SUITE(eosio_distribute_tests)
+
+BOOST_AUTO_TEST_CASE( deploy_config ) try {
+    eosio_distribute_tester t;
+    
+    // configure checks
+    BOOST_REQUIRE_EQUAL(t.error("missing authority of eosio.dist"), t.setdistrib(sys, {}));
+    BOOST_REQUIRE_EQUAL(t.wasm_assert_msg("Cannot set account to eosio.dist"), t.setdistrib(dist_acct, { {dist_acct, 10000} }));
+    BOOST_REQUIRE_EQUAL(t.wasm_assert_msg("Account does not exist: alice"), t.setdistrib(dist_acct, { {alice, 10000} }));
+    t.create_accounts_with_resources({alice, bob}, sys);
+    BOOST_REQUIRE_EQUAL(t.wasm_assert_msg("Total percentage does not equal 100%"), t.setdistrib(dist_acct, { {alice, 1000} }));
+    // force overflow
+    uint16_t pct = 65535;
+    // pct++;
+    BOOST_REQUIRE_EQUAL(t.wasm_assert_msg("Only positive percentages are allowed"), t.setdistrib(dist_acct, { {alice, ++pct} }));
```

Comment:
The overflow has nothing to do with the contract's behavior.  The value that is serialized in the action data is 0.